### PR TITLE
Start enforcing Ruff rule ANN201 in `modeling`

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

[ANN201](https://docs.astral.sh/ruff/rules/missing-return-type-undocumented-public-function/) checks for public functions without a return type annotation. `modeling` was the last sub-package where this rule was not enforced. Note that Ruff is configured to check this rule only for functions that already have a type annotation in their signature.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/modeling/bounding_box.py
Comment (neutrinoceros) [inline astropy/modeling/bounding_box.py:1607]: I'm not 100% on this one: it looks like the return type is by construction the same as in `_fix_input_selector_arg`, which is currently not annotated and appears to unconditionally return a `CompoundBoundingBox` instance. Am I missing something ?

**Review by WilliamJamieson — APPROVED** [review]

File: astropy/modeling/bounding_box.py
Comment (eerovaher) [reply astropy/modeling/bounding_box.py:1607]: I wrote down these return types more than two months ago and I don't remember what my reasoning was. This particular function has a docstring, but it does not document what the return type is. Many of the functions called here have neither a docstring nor any annotations. It is possible the annotation here isn't right.

Comment (pllim) [reply astropy/modeling/bounding_box.py:1607]: @WilliamJamieson , can you please confirm whether `Self` is correct here? Thanks!

**Comment by eerovaher** [comment]
Rebased as requested by https://groups.google.com/g/astropy-dev/c/2y2vVj1go0w

File: astropy/modeling/bounding_box.py
Comment (pllim) [reply astropy/modeling/bounding_box.py:1611]: Is this conversation resolved?

Comment (eerovaher) [reply astropy/modeling/bounding_box.py:1611]: I rebased the pull request as it was. I have not rechecked the correctness of any of the annotations.

**Comment by nstarman** [comment]
@pllim, could you merge this in? This PR has the requisite approvals, but the milestone checker hasn't reported back. Thanks!

**Comment by pllim** [comment]
FYI -- I simply removed the milestone, wait for the ❌ , and then re-applied. Anyone with write access could do that.

**Pull request merged**
